core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockReader.java

```diff
@@ -230,12 +229,13 @@ public int transferTo(ByteBuf buf) throws IOException {
       try {
         bufCopy.writerIndex(buf.writerIndex());
         while (bufCopy.readableBytes() > 0) {
+          mLocalBlockStore.requestSpace(mBlockMeta.getSessionId(), mBlockMeta.getBlockId(),
+              mInStreamPos - mBlockWriter.getPosition());
           mBlockWriter.transferFrom(bufCopy);
         }
       } catch (Exception e) {
-        LOG.warn("Failed to cache data read from UFS: %s", e.getMessage());
-        mBlockWriter.close();
-        mBlockWriter = null;
+        LOG.warn("Failed to cache data read from UFS on transferTo(): {}", e.getMessage());
```

Comment:
what does transferTo() mean in terms that a user unfamiliar with the code would understand?

Reply:
well essentially "read" is the same as "transferTo" but with different argument type.